snapshottest/module.py

```diff
@@ -17,6 +18,14 @@ def _escape_quotes(text):
     return text.replace("'", "\\'")
 
 
+def _load_source(module_name, filepath):
+    spec = importlib.util.spec_from_file_location(module_name, filepath)
+    module = importlib.util.module_from_spec(spec)
+    sys.modules[module_name] = module
```

Comment:
Is it necessary to write to sys.modules for this to work?

Reply:
No, writing to `sys.modules` is not required for this to work (it just improves performance if the same python module is loaded multiple times)

Reply:
@paulmelnikow I'm not the owner of the PR, but I've found a similar problem and question on official Python mailing lists: https://discuss.python.org/t/how-do-i-migrate-from-imp/27885

It looks like writing to sys.modules is indeed the correct, see [this answer](https://discuss.python.org/t/how-do-i-migrate-from-imp/27885/9)

Reply:
I guess my ask here is that it would be good to add some links to the imp documentation, mailing list, etc. to explain this implementation, which otherwise seems somewhat peculiar.